### PR TITLE
Adding util.cpp and util.h files

### PR DESCRIPTION
**Pull request by Grant-Postma**
Moved the common functions from multiple classes to a central utilities file, so that this code does not need to reside in multiple locations, reducing confusion and hindering future iterations to the methods.

### CODE REVIEW

File: demos/TicTacToe/Classes/MainMenuScene.cpp [hunk 1]
```diff
@@ -16,6 +16,8 @@
 
 #include <regex>
 
+#include "cocos2d.h"
```
Comment (DellaBitta) [inline demos/TicTacToe/Classes/MainMenuScene.cpp:19]: The cocoas2d.h include is already in the header, so it's not needed here, too.

File: demos/TicTacToe/Classes/MainMenuScene.cpp [hunk 2]
```diff
@@ -16,6 +16,8 @@
...
+#include "util.h"
 #include "TicTacToeScene.h"
```
Comment (DellaBitta) [inline demos/TicTacToe/Classes/MainMenuScene.cpp:21]: should probably have consistency of capitalization in the file names.

File: demos/TicTacToe/Classes/TicTacToeScene.h [hunk 3]
```diff
@@ -18,6 +18,8 @@
 #include "MainMenuScene.h"
 #include "TicTacToeLayer.h"
 #include "cocos2d.h"
+#include "util.h"
+
```
Comment (DellaBitta) [inline demos/TicTacToe/Classes/TicTacToeScene.h:22]: remove this new extra line.

File: demos/TicTacToe/Classes/TicTacToeLayer.h [hunk 4]
```diff
@@ -27,9 +27,10 @@
 #include <string>
 #include <unordered_set>
 
-#include "MainMenuScene.h"
 #include "TicTacToeScene.h"
 #include "cocos2d.h"
+#include "util.h"
+
```
Comment (DellaBitta) [inline demos/TicTacToe/Classes/TicTacToeLayer.h:33]: remove this new extra line.

File: demos/TicTacToe/Classes/util.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef TICTACTOE_DEMO_CLASSES_UTIL_H_
+#define TICTACTOE_DEMO_CLASSES_UTIL_H_
+
+#include <random>
```
Comment (DellaBitta) [inline demos/TicTacToe/Classes/util.h:17]: Please go through these headers and determine those which are required for this header and those that need to be in the cpp file only, pruning those that are unnecessary.  For instance I don't *think* that you need random.h or cocos2d.h, or most of the firebase headers (except the one that defines the futures).

File: demos/TicTacToe/Classes/MainMenuScene.cpp (hunk 1 above)
Comment (DellaBitta) [reply demos/TicTacToe/Classes/MainMenuScene.cpp:19]: Uh, unless you abide my other comment in the header, then it should remain here.  You'll see.  No spoilers!

File: demos/TicTacToe/Classes/util.h (hunk 5 above)
Comment (DellaBitta) [reply demos/TicTacToe/Classes/util.h:17]: The reason that we do this is to reduce compiler times (along with other reasons).  Any file that includes this header will require the at the toolchain append all of these other header files. But if they're not needed here then that's adding cost in time for no true reason.

File: demos/TicTacToe/Classes/MainMenuScene.cpp (hunk 1 above)
Comment (Grant-Postma) [reply demos/TicTacToe/Classes/MainMenuScene.cpp:19]: Going to follow up with another pull request to clean up the import/using statements

File: demos/TicTacToe/Classes/MainMenuScene.cpp (hunk 2 above)
Comment (Grant-Postma) [reply demos/TicTacToe/Classes/MainMenuScene.cpp:21]: The util.h will match the updated naming convention to follow google's style guide

**Pull request merged**